### ISSUE TITLE
Error: There must be something wrong about the Peak_method value in your primary params set !

### ISSUE THREAD

**Issue by grangel0955** [xia-lab/MetaboAnalystR#183]
When trying to optimize parameters I keep getting the following error. data_QC is the directory path to .mzXML files. 

>raw_data <- PerformDataTrimming(data_QC, rt.idx = 1.0) 
> param_initial <- SetPeakParam(platform = "HPLC-Orbitrap")
> param_optimized <- PerformParamsOptimization(raw_data, param = param_initial)        
>[1] "DoE Optimization Starting Now!"
>'ncore' is absent, will use 2/3 CPU threads of total!
>[1] "Evaluating Noise level..."
>Error in optimize.xcms.doe(raw_data, param = param, ncore = ncore) :       
 > There must be something wrong about the Peak_method value in your primary params set !

I am working in a RHEL7 environment on R 4.0.5. 

Any help would be greatly appreciated!

**Comment by grangel0955** [comment]
I was able to get the PerformParamsOptimization function to continue if I set Peak_method = centWave and RT_method = obiwarp. However, I now get an error later in the function (see below). I cannot get it to work at all with centWave and loess for Peak and RT methods respectively. 

Model Parsing...
Error in lm.fit(x, y, offset = offset, singular.ok = singular.ok, ...) : 
  0 (non-NA) cases
Calls: PerformParamsOptimization ... createModel -> <Anonymous> -> eval -> eval -> lm -> lm.fit
In addition: Warning message:
In Noise_evaluate(raw_data) :
  You are strongly advised to use wider rt.idx or more samples or other trimming strategy !
Execution halted

Any help on this?

**Comment by coltonlloyd** [comment]
@grangel0955 I was getting the same error. It looks like the default parameter "lowess" is not recognized as an `RT_method`. Setting `RT_method` to "peakgroup" corrected this issue.

However, I ran into another issue when trying to run PerformParamsOptimization with `ncore>1`. The `optimize_function_list` variable is not defined anywhere in `parameters_optimization.R` source code, which caused another error.

PerformParamsOptimize appears to run if it is not in parallel. 

Hope this helps!

**Comment by grangel0955** [comment]
I was able to get the function to finish without the error during Model Parsing if I loaded the OptiLCMS package after the MetaboAnalystR package. It appears there my be something wrong with the function in the MetaboAnalystR package.

**Comment by coltonlloyd** [comment]
That works for me as well. Thanks for the tip

**Comment by demonlj** [comment]
using library(OptiLCMS) or require(OptiLCMS) could solve this problem, however, causing another problem in 
```
# Import raw MS data. The "SetPlotParam" parameters can be used to determine plot or not.
rawData <- ImportRawMSData(data_folder_Sample,plotSettings = SetPlotParam(Plot=T))

# Peak Profiling with optimized parameters
mSet <- PerformPeakProfiling(rawData,param_optimized$best_parameters,
                             plotSettings = SetPlotParam(Plot = T))
```
error code:
```
> rawData <- ImportRawMSData(data_folder_Sample,plotSettings = SetPlotParam(Plot=T))
Something wierd happened, don't worry. We will re-initialize one automatically !
Error in Path2Files(path = path) : Multiple data formats are not allowed!
> unloadNamespace("OptiLCMS")
> rawData <- ImportRawMSData(data_folder_Sample,plotSettings = SetPlotParam(Plot=T))
[1] "The number of CPU cores to be used is set to 6."
[1] "To reduce memory usage BPIS and TICS plots will be created using only 10 samples per group."
[1] "Plotting BPIS and TICS."
[1] "Successfully imported raw MS data!"
```
This must be due to the conflict between the two packages(MetaboAnalystR and OptiLCMS), hoping it could be solved latter.

**Comment by Zhiqiang-PANG** [comment]
Yes, Please library(OptiLCMS) if you want to use MetaboAnalystR to process raw data. We will coordinate these two packages in the coming version 4. I am closing this for now.
